dev_faq.md

```diff
@@ -3,6 +3,118 @@ layout: post
 title: Developer FAQ
 ---
 
+- [Is there a BIDS App template?](#is-there-a-bids-app-template)
+- [Which container should I use to start building my BIDS App?](#which-container-should-i-use-to-start-building-my-bids-app)
+- [API](#api)
+    - [What should the API of my BIDS App look like?](#what-should-the-api-of-my-bids-app-look-like)
+    - [Can I add more arguments to the API of my App?](#can-i-add-more-arguments-to-the-api-of-my-app)
+    - [What the kinds of analysis levels (`participant` and `group`) mean?](#what-the-kinds-of-analysis-levels-participant-and-group-mean)
+    - [What do we do if our application does not have any use for the group level analysis?](#what-do-we-do-if-our-application-does-not-have-any-use-for-the-group-level-analysis)
+- [Is it mandatory to first check the dataset validity using the BIDS-validator?](#is-it-mandatory-to-first-check-the-dataset-validity-using-the-bids-validator)
+- [How do I upload my BIDS App to the BIDS App Github org?](#how-do-i-upload-my-bids-app-to-the-bids-app-github-org)
+- [Testing Data](#testing-data)
+- [Versioning](#versioning)
+    - [When is a new image deposited to Docker Hub?](#when-is-a-new-image-deposited-to-docker-hub)
+    - [How to tag a new release?](#how-to-tag-a-new-release)
+    - [How should I version my BIDS App?](#how-should-i-version-my-bids-app)
+    - [I want to release a new version of a BIDS App, but the pipeline version is the same?](#i-want-to-release-a-new-version-of-a-bids-app-but-the-pipeline-version-is-the-same)
+    - [Where should I describe changes between versions?](#where-should-i-describe-changes-between-versions)
+    - [How can I check a version of a container I have available locally?](#how-can-i-check-a-version-of-a-container-i-have-available-locally)
+    - [How can I download a particular version of a BIDS App?](#how-can-i-download-a-particular-version-of-a-bids-app)
+
+## Is there a BIDS App template?
+
+Have a look at the
+[example BIDS App repository](https://github.com/BIDS-Apps/example). A
+minimalist example of a BIDS App consisting of a Dockerfile and a simple entry
+point script (written in this case in Python) accepting the standard BIDS Apps
+command line arguments.
+
+## Which container should I use to start building my BIDS App?
+
+The only minimum requirements of a BIDS App's container is its ability to run
+your pipeline. So for example, if your App is mostly Python based it should be
+sufficient to start with any image that has Python and include your environment
+dependencies.
+
+## API
+
+### What should the API of my BIDS App look like?
+
+The obligatory arguments of the API of any BIDS App are:
+
+- `bids_dir`
+- `output_dir`
+- `analysis_level`
+
+A typical BIDS App call would thus look like
+
+```bash
+$ bids_dir=/home/jane/input/ds00101
+
+$ output_dir=/home/jane/output/
+
+$ analysis_level=participant
+
+$ BIDSapp_name $bids_dir $output_dir $analysis_level
+```
```

Comment:
This feels a bit silly, no?

Reply:
agreed, using the $variables only complicates this example

Reply:
on the other hand, I am also fine with it as is ... no biggie

Reply:
updated it.